### PR TITLE
Fix sign comparison

### PR DESCRIPTION
**Pull request by r-barnes**
Differential Revision: D29885396

### CODE REVIEW

File: aten/src/ATen/native/cuda/CuFFTPlanCache.h [hunk 1]
```diff
@@ -507,7 +508,7 @@ class CuFFTParamsLRUCache {
     // first.
     TORCH_CHECK(new_size >= 0,
              "cuFFT plan cache size must be non-negative, but got ", new_size);
-    TORCH_CHECK(new_size <= CUFFT_MAX_PLAN_NUM,
+    TORCH_CHECK(new_size <= static_cast<int64_t>(CUFFT_MAX_PLAN_NUM),
```
Comment (albanD) [inline aten/src/ATen/native/cuda/CuFFTPlanCache.h:511]: Should we just make CUFFT_MAX_PLAN_NUM a int64_t from the beginning? There is a static assert that it is always positive already.

File: torch/csrc/autograd/custom_function.h [hunk 2]
```diff
@@ -315,12 +315,10 @@ variable_list CppNode<T>::apply(variable_list&& inputs) {
 
   auto outputs = T::backward(&ctx_, backward_inputs);
 
-  // NOLINTNEXTLINE(cppcoreguidelines-init-variables)
-  int num_forward_inputs = is_variable_input_.size();
-  auto num_outputs = outputs.size();
+  const auto num_forward_inputs = static_cast<int64_t>(is_variable_input_.size());
```
Comment (albanD) [inline torch/csrc/autograd/custom_function.h:318]: Why can't these two stay size_t?

Comment (r-barnes) [reply torch/csrc/autograd/custom_function.h:318]: `num_outputs` comes from `auto outputs = T::backward(&ctx_, backward_inputs);` where `T` is a template. I suspect the type of `outputs.size()` varies for different `T`.

File: aten/src/ATen/native/cuda/CuFFTPlanCache.h (hunk 1 above)
Comment (r-barnes) [reply aten/src/ATen/native/cuda/CuFFTPlanCache.h:511]: I'd support that. Signed types seem to be generally easier to work with; however, I suspect we'll still have inappropriate comparisons. Let me know what works best for you.

File: torch/csrc/autograd/custom_function.h (hunk 2 above)
Comment (albanD) [reply torch/csrc/autograd/custom_function.h:318]: Ho interesting. In my mind we were enforcing that backward always returns a variable_list. But that is not the case indeed.

File: aten/src/ATen/native/cuda/CuFFTPlanCache.h (hunk 1 above)
Comment (albanD) [reply aten/src/ATen/native/cuda/CuFFTPlanCache.h:511]: Given that these values can never really overflow a signed int. I would lean towards whatever reduces the number of casts the most :D
But if you think this is cleaner, I'm happy with that as well.

**Review by albanD — APPROVED** [review]

**Pull request closed without merging**